utils/accountprofile.go

```diff
@@ -128,7 +128,7 @@ func (aP *AccountProfile) Clone() (acnt *AccountProfile) {
 		}
 	}
 	if aP.Balances != nil {
-		acnt.Balances = make([]*Balance, len(aP.Balances))
+		acnt.Balances = make(map[string]*Balance, len(aP.Balances))
 		for i, value := range aP.Balances {
```

Comment:
As best practice, please use "k" for key and "i" when dealing with slice indexes.